pkg/cleaner/jobcontroller.go

```diff
@@ -75,10 +74,20 @@ func NewCleanupJob(bd *v1alpha1.BlockDevice, volMode VolumeMode, tolerations []v
 	mountName := "vol-mount"
 
 	if volMode == VolumeModeBlock {
+		jobContainer.Command = []string{"/bin/sh", "-c"}
+
+		pathRegex := bd.Spec.Path
+		// if deviceType is a disk, there can be partitions on the disk. In that case
+		// partitions need to be wiped before wiping main disk
+		if bd.Spec.Details.DeviceType == "disk" {
+			pathRegex = bd.Spec.Path + "*"
+		}
 		// wipefs erases the filesystem signature from the block
 		// -a    wipe all magic strings
 		// -f    force erasure
-		jobContainer.Command = getCommand(BlockCleanerCommand, "-af", bd.Spec.Path)
+		args := fmt.Sprintf("ls -r %s | xargs wipefs -af", pathRegex)
```

Comment:
Can the ls command be made more specific for partitions? For example, what if we have the case like: `/dev/sda` and `/dev/sdaa` .  A case where mounted volumes (like iSCSI) start getting a new device mount.

Reply:
Will fix it @kmova

Reply:
Done.